src/hooks/useTokenPrice.ts

```diff
@@ -0,0 +1,40 @@
+import useSWR from "swr"
+
+import { CHAIN_ID } from "@/constants"
+
+const useTokenPrice = tokenList => {
+  const fetchPrice = async () => {
+    try {
+      const tokens = tokenList.filter(token => token.chainId === CHAIN_ID.L1 && token.address).map(token => token.address)
+      const tokenAddresses = tokens.join("%2C")
+
+      const fetchEthPrice = fetch(`https://api.coingecko.com/api/v3/simple/price?ids=ethereum&vs_currencies=usd`)
+      const fetchErc20Price = fetch(
+        `https://api.coingecko.com/api/v3/simple/token_price/ethereum?contract_addresses=${tokenAddresses}&vs_currencies=usd`,
+      )
+
+      const [ethPriceResponse, erc20PriceResponse] = await Promise.all([fetchEthPrice, fetchErc20Price])
```

Comment:
can we use `Promise.allSettled` to avoid a situation where one failure causes all to fail?